parser-typechecker/tests/Unison/Core/Test/Name.hs

```diff
@@ -4,18 +4,26 @@ module Unison.Core.Test.Name where
 
 import           EasyTest
 import           Unison.Name                   as Name
+import           Unison.NameSegment            as NameSegment
+import           Data.List                      ( intercalate )
+import           Data.Text                      ( pack )
 
 test :: Test ()
-test = scope "name" $ tests [
-  scope "suffixes" $
-    tests
-      [ scope "empty" $ expectEqual (suffixes "") []
-      , scope "one namespace" $ expectEqual (suffixes "bar") ["bar"]
-      , scope "two namespaces"
-        $ expectEqual (suffixes "foo.bar") ["foo.bar", "bar"]
-      , scope "multiple namespaces"
-        $ expectEqual (suffixes "foo.bar.baz") ["foo.bar.baz", "bar.baz", "baz"]
-      , scope "terms named `.`"
-        $ expectEqual (suffixes "base..") ["base..", "."]
-      ]
+test = scope "name" $ tests
+  [ scope "suffixes" $ tests
+    [ scope "empty" $ expectEqual (suffixes "") []
+    , scope "one namespace" $ expectEqual (suffixes "bar") ["bar"]
+    , scope "two namespaces"
+      $ expectEqual (suffixes "foo.bar") ["foo.bar", "bar"]
+    , scope "multiple namespaces"
+      $ expectEqual (suffixes "foo.bar.baz") ["foo.bar.baz", "bar.baz", "baz"]
+    , scope "terms named `.`" $ expectEqual (suffixes "base..") ["base..", "."]
+    ]
+  , scope "segments" $ do
+    numDots <- int' 0 10
+    numSegs <- int' 0 10
+    n       <- int' 0 10
+    segs <- listOf n . pick $ replicate numDots "." ++ replicate numSegs "foo"
+    expectEqual (segments $ Name . pack $ intercalate "." segs)
+                (NameSegment . pack <$> segs)
```

Comment:
I believe this is a comprehensive test for `segments`

Reply:
What is this test testing?  Are the `segs` things like `"...foofoo"`?

Reply:
segs is a list of elements which are either `.` or `foo`. So it generates paths like `foo...foo.foo......`

Reply:
It's testing that a rountrip from a list to a string back to a list via `segments` is the identity.

Reply:
Ok, sorry, just learning some Haskell here; it's picking `n` segments out of a distribution with a ratio of `numDots`:`numSegs` dots to "foo"s?

Reply:
That's the one.